File: ios/Packages/Primitives/Sources/WalletImportResult.swift
// Copyright (c). Gem Wallet. All rights reserved.

import Primitives

public enum WalletImportResult: Sendable {
    case new(Wallet)
    case existing(Wallet)

    public var wallet: Wallet {
        switch self {
        case let .new(wallet), let .existing(wallet):
            wallet
        }
    }
}

File: android/gemcore/src/main/kotlin/com/gemwallet/android/application/wallet_import/values/WalletImportResult.kt
package com.gemwallet.android.application.wallet_import.values

import com.wallet.core.primitives.Wallet

sealed class WalletImportResult {
    abstract val wallet: Wallet

    data class New(override val wallet: Wallet) : WalletImportResult()
    data class Existing(override val wallet: Wallet) : WalletImportResult()
}
